src/components/Catalog.vue

```diff
@@ -99,7 +99,7 @@ export default defineComponent({
   },
   data () {
     return {
-      activeView: 'grid'
+      activeView: 'grid' as 'grid' | 'table'
```

Comment:
https://www.typescriptlang.org/docs/handbook/literal-types.html#string-literal-types could we define it as a string literal?

Reply:
TS will always type a string literal as a `string` rather than a more specific type like its actual literal value `'grid'` or a string union `'grid' | 'table'`. So any code that depends on something more specific down-the-line will not be aware of a literal type.

So in this case, it is as if the `data` function is returning `{ activeView: string }` even though TS can technically see it is `{ activeView: 'grid' }` (it will always choose `string`). So the three ways to satisfy later code that relies on a type like `'grid'` or `'grid' | 'table'` are:
1. Use `as 'grid'` to tell TS that you don't just won't a `string` but a specific literal
2. Annotate the type somewhere, in this case we could do `data (): { activeView: 'grid' | 'table' } {` on line 102 to avoid having to use type coercion with `as`
3. Do a check down-the-line that `data.activeView === 'grid' || data.activeView === 'table'` to narrow the type safely without using `as`.

1 & 2 make sense for this case, and 3 does not.

I like option 2 better because it use explicit type annotations rather than coercion with `as`, but it ends up with the same types.